tests/Microsoft.Azure.CosmosEventSourcingTests/Extensions/ObjectExtensions.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Reflection;
+
+namespace Microsoft.Azure.CosmosEventSourcingTests.Extensions;
+
+public static class ObjectExtensions
+{
+    public static void SetPrivatePropertyValue<T>(this object obj, string propName, T val)
```

Comment:
Code smell, why are we doing reflection-based `private` property setting? There are other ways to do this more efficiently and reliably. For example, `internal` and then you could set the visibility to this lib.

Reply:
Yes i totally agree, going to do some refactoring of the EventItem to make this no an issue, just had to wait while the other PR was merged to avoid conflicts, plan to sort this within the day 👍🏼